File: java/jcf/Queue.java
package BuiltIn;

import java.util.LinkedList;
import java.util.Queue;

public class Queue_ {

    public static void main(String[] args) {
        Queue<Integer> queue = new LinkedList<>();

        // Enqueue elements
        queue.offer(1);
        queue.offer(2);
        queue.offer(3);

        // or can do .add() to add to queue
        queue.add(3);
        queue.add(2);

        // Dequeue elements
        int firstElement = queue.poll();
        System.out.println(firstElement); // Output: 1

        // Peek at the front element
        int frontElement = queue.peek();
        System.out.println(frontElement); // Output: 2

        // Check if the queue is empty
        boolean isEmpty = queue.isEmpty();
        System.out.println(isEmpty); // Output: false

        // Get the size of the queue
        int size = queue.size();
        System.out.println(size); // Output: 2
    }
}

File: cpp/stl/queue.cpp
#include <iostream>
#include <queue>

int main() {
    // Create a queue of integers
    std::queue<int> myQueue;

    // Enqueue elements into the queue
    myQueue.push(10);
    myQueue.push(20);
    myQueue.push(30);

    // Print the front element of the queue
    std::cout << "Front element of the queue: " << myQueue.front() << std::endl;

    // Dequeue elements from the queue
    myQueue.pop();

    // Print the front element after dequeuing
    std::cout << "Front element after dequeuing: " << myQueue.front() << std::endl;

    // Check if the queue is empty
    std::cout << "Is queue empty? " << (myQueue.empty() ? "Yes" : "No") << std::endl;

    // Get the size of the queue
    std::cout << "Size of the queue: " << myQueue.size() << std::endl;

    return 0;
}

